File: samples/java-markdown/app/src/main/java/connector/GraphService.java
package connector;

import java.io.IOException;
import java.util.Properties;

import com.azure.identity.ClientSecretCredential;
import com.azure.identity.ClientSecretCredentialBuilder;
import com.microsoft.graph.authentication.TokenCredentialAuthProvider;
import com.microsoft.graph.httpcore.HttpClients;
import com.microsoft.graph.requests.GraphServiceClient;

import okhttp3.OkHttpClient;

public class GraphService {
    private static GraphServiceClient<okhttp3.Request> client;

    public static GraphServiceClient<okhttp3.Request> getClient() throws IOException {
        if (client == null) {
            final Properties properties = new Properties();
            properties.load(GraphService.class.getResourceAsStream("application.properties"));

            final ClientSecretCredential credential = new ClientSecretCredentialBuilder()
                    .clientId(properties.getProperty("app.clientId"))
                    .clientSecret(properties.getProperty("app.clientSecret"))
                    .tenantId(properties.getProperty("app.tenantId"))
                    .build();
            final TokenCredentialAuthProvider authProvider = new TokenCredentialAuthProvider(credential);
            final OkHttpClient okHttpClient = HttpClients.createDefault(authProvider)
                    .newBuilder()
                    .addInterceptor(new CompleteJobWithDelayHandler(60000))
                    // .addInterceptor(new DebugHandler())
                    .callTimeout(25, java.util.concurrent.TimeUnit.MINUTES)
                    .build();

            client = GraphServiceClient.builder()
                    .httpClient(okHttpClient)
                    .buildClient();
        }

        return client;
    }
}

File: samples/dotnet-csharp-m365-roadmap/GraphService.cs
using Azure.Identity;
using Microsoft.Graph;
using Microsoft.Extensions.Configuration;

public class GraphService
{
  static GraphServiceClient? _client;
  public static GraphServiceClient Client
  {
    get
    {
      if (_client is null)
      {
        var builder = new ConfigurationBuilder().AddUserSecrets<GraphService>();
        var config = builder.Build();

        var clientId = config["EntraId:ClientId"];
        var clientSecret = config["EntraId:ClientSecret"];
        var tenantId = config["EntraId:TenantId"];

        var credential = new ClientSecretCredential(tenantId, clientId, clientSecret);
        _client = new GraphServiceClient(credential);
      }

      return _client;
    }
  }
}